apps/condo/pages/auth/change-password.tsx

```diff
@@ -1,158 +1,172 @@
-                }, () => {
-                    auth.refetch().then(() => {
-                        setIsSaving(false)
-                        Router.push('/')
-                    })
-                })
-            },
-            intl,
-            form,
-        }).catch(() => {
-            setIsSaving(false)
-        })
-    }
-
-    const onResetPasswordClick = useCallback(() => Router.push('/auth/forgot'), [])
-
-    const [checkConfirmPhoneActionToken] = useLazyQuery(GET_PHONE_BY_CONFIRM_PHONE_TOKEN_QUERY, {
-        onError: error => {
-            setRecoveryTokenError(error)
-            setIsLoading(false)
-        },
-        onCompleted: () => {
-            setRecoveryTokenError(null)
+    }), [PleaseConfirmYourPasswordMessage, PleaseInputYourPasswordMessage, TwoPasswordDontMatchMessage, changeMessage, minPasswordLengthValidator, requiredValidator])
+
+    const authWithPhoneAndPassword = useCallback(async (phone: string, password: string): Promise<void> => {
+        try {
+            const sender = getClientSideSenderInfo()
+            const captcha = await executeCaptcha()
+            const res = await authenticateUserWithPhoneAndPasswordMutation({
+                variables: {
+                    data: {
+                        dv: 1,
+                        sender,
+                        captcha,
+                        password,
+                        phone,
+                        userType: UserType.Staff,
+                    },
+                },
+            })
+
+            if (res.errors || !res?.data?.result?.item?.id) {
+                const error: Error & { originalErrors? } = new Error('Authorization failed (request)')
+                error.originalErrors = res.errors
+                throw error
+            }
+        } catch (error) {
+            console.error('Authorization failed')
+            console.error(error)
+            throw error
+        }
+    }, [authenticateUserWithPhoneAndPasswordMutation, executeCaptcha])
+
+    const onSubmit = useCallback(async (values: typeof INITIAL_VALUES) => {
+        if (isLoading) return
+
+        setIsLoading(true)
+
+        try {
+            const { password } = values
+            const sender = getClientSideSenderInfo()
+            // const captcha = await executeCaptcha()
+
+            const res = await changePasswordWithTokenMutation({
+                variables: {
+                    data: {
+                        dv: 1,
+                        sender,
+                        password,
+                        token,
+                        // captcha,
+                    },
+                },
+            })
+
+            const status = res?.data?.result?.status
+            const phone = res?.data?.result?.phone
+            if (!res.errors && status === 'ok' && phone) {
+                await authWithPhoneAndPassword(phone, password)
+                await refetch()
+                await Router.push('/')
```

Comment:
why not router from useRouter?

Reply:
It's easier this way

Reply:
What's easier?

Reply:
```
import Router from 'next/router'

await Router.push('/')
```

vs 

```
import { useRouter } from 'next/router'

const router = useRouter()

await router.push('/')
```

Reply:
The code becomes a little shorter, simpler.
Also, we don't need to specify extra dependencies when using `useCallback`